admin-dev/themes/new-theme/js/pages/catalog-price-rule/form/index.ts

```diff
@@ -42,16 +42,29 @@ $(() => {
       // Reduction Amount
       const reductionTypeSelect = document.querySelector<HTMLSelectElement>(CatalogPriceRuleFormMap.reductionTypeSelect);
 
-      if (reductionTypeSelect?.options[reductionTypeSelect.selectedIndex].value === 'amount') {
-        const reductionTypeAmountSymbols = document.querySelectorAll(
-          CatalogPriceRuleFormMap.reductionTypeAmountSymbol,
-        );
+      if (reductionTypeSelect) {
+        // Update the amount option innerHTML
+        for (let i = 0; i < reductionTypeSelect.options.length; i += 1) {
+          const reductionOption = reductionTypeSelect.options[i];
 
-        if (reductionTypeAmountSymbols.length) {
-          reductionTypeAmountSymbols.forEach((value: Element) => {
-            const elt = value;
-            elt.innerHTML = symbol;
-          });
+          if (reductionOption.value === 'amount') {
+            reductionOption.innerHTML = symbol;
+          }
+        }
+
+        const selectedReduction = reductionTypeSelect.options[reductionTypeSelect.selectedIndex].value;
+
+        if (selectedReduction === 'amount') {
```

Comment:
Yeah and this whole code needs refacto anyway It's 90% similar to the code for specific price And there are two/three components that could probably be merged into on better written Because they are only used in this case anyway, and splitting them only brought duplicated split code anyway